plugins/outputs/http/http.go

```diff
@@ -0,0 +1,273 @@
+	return sampleConfig
+}
+
+// Writes metrics over HTTP POST
+func (h *Http) Write(metrics []telegraf.Metric) error {
+	if err := validate(h); err != nil {
+		return err
+	}
+
+	for _, metric := range metrics {
+		buf, err := h.serializer.Serialize(metric)
+
+		if err != nil {
+			return fmt.Errorf("E! Error serializing some metrics: %s", err.Error())
+		}
+
+		h.requestBodyMetricBuffer = append(h.requestBodyMetricBuffer, buf)
+
+		if h.BufferLimit <= len(h.requestBodyMetricBuffer) {
+			requestBody, err := h.makeRequestBody()
+
+			if err != nil {
+				return fmt.Errorf("E! Error serialized metric is not assembled : %s", err.Error())
+			}
+
+			response, err := h.write(requestBody)
+
+			if err := h.isOk(response, err); err != nil {
+				return err
+			}
+
+			defer response.Body.Close()
+		}
+	}
+
+	return nil
+}
+
+func (h *Http) isOk(response *http.Response, err error) error {
+	if response == nil || err != nil {
+		return fmt.Errorf("E! %s request failed! %s.", h.URL, err.Error())
+	}
+
+	if !h.isExpectedStatusCode(response.StatusCode) {
+		return fmt.Errorf("E! %s response is unexpected status code : %d.", h.URL, response.StatusCode)
+	}
+
+	return nil
+}
+
+func (h *Http) isExpectedStatusCode(responseStatusCode int) bool {
+	if h.expectedStatusCodeMap == nil {
+		h.expectedStatusCodeMap = make(map[int]bool)
+
+		for _, expectedStatusCode := range h.ExpectedStatusCodes {
+			h.expectedStatusCodeMap[expectedStatusCode] = true
+		}
+	}
+
+	if h.expectedStatusCodeMap[responseStatusCode] {
+		return true
+	}
+
+	return false
+}
+
+// required option validate
+func validate(h *Http) error {
+	if h.URL == "" || len(h.HttpHeaders) == 0 || len(h.ExpectedStatusCodes) == 0 || h.BufferLimit == 0 {
+		return errors.New("E! Http ouput plugin is not working. Because your configuration omits the required option. Please check url, http_headers, expected_status_codes, buffer_limit is empty!")
+	}
+
+	return nil
+}
+
+func (h *Http) write(buf []byte) (*http.Response, error) {
+	req, err := http.NewRequest(POST, h.URL, bytes.NewBuffer(buf))
+
+	for _, httpHeader := range h.HttpHeaders {
+		keyAndValue := strings.Split(httpHeader, ":")
+		req.Header.Set(keyAndValue[0], keyAndValue[1])
+	}
+
+	req.Close = true
+	req.WithContext(h.cancelContext)
+
+	response, err := h.client.Do(req)
+
+	return response, err
+}
+
+// makeRequestBody does additional work so that each serializer can include serialized metrics in the request body.
+// For example, a serialized metric in json.JsonSerializer returns a metric that looks like JsonObject {"key1": "value1"}.
+// In order to accumulate this in the buffer and send it to the request body at once, you need to convert it to Array Json Object [{"key1": "value1"}, {"key2": "value2"}].
+// Thus, makeRequestBody works by making the request body contain the metric returned by each serializer.
+func (h *Http) makeRequestBody() ([]byte, error) {
+	switch h.serializer.(type) {
+	case *json.JsonSerializer:
```

Comment:
The content-type of http determines the format of the data to be included in the request body. For example, if it is application / json, the data must be converted to match that spec.
Currently, json.jsonSerializer returns a json object {}. If you put it in buf and simply send it to the request body, a problem occurs.
{"Key1", "value1"}, {"key2": "value2"} (wrong)
[{"Key1", "value1"}, {"key2": "value2"}] (right)
This is because several Json objects must be represented by an Array. So I thought I needed to collect several Json objects and create data to fit in the request body in Array Json Object format. 
So, in order to handle plain / text and json differently, I made the code like above.